src/tcgwars/logic/impl/gen4/HeartgoldSoulsilverPromos.groovy

```diff
@@ -177,383 +198,258 @@ public enum HeartgoldSoulsilverPromos implements LogicCardInfo {
             text "80 damage. Discard 2 Energy attached to Latios."
             energyCost C, C, C, C
             attackRequirement {}
             onAttack {
-              damage 0
+              damage 80
+              discardSelfEnergyAfterDamage(C, C)
             }
           }
-
         };
       case CLEFFA_HGSS12:
-        return basic (this, hp:HP030, type:COLORLESS, retreatCost:0) {
-          pokeBody "Sweet Sleeping Face", {
-            text "As long as Cleffa is Asleep, prevent all damage done to Cleffa by attacks."
-            delayedA {
-            }
-          }
-          move "Eeeeeeek", {
-            text "Shuffle your hand into your deck, then draw 6 cards. Cleffa is now Asleep."
-            energyCost ()
-            attackRequirement {}
-            onAttack {
-              damage 0
-            }
-          }
-
-        };
+        return copy (HeartgoldSoulsilver.CLEFFA_17, this);
       case SMOOCHUM_HGSS13:
-        return basic (this, hp:HP030, type:PSYCHIC, retreatCost:0) {
-          pokeBody "Sweet Sleeping Face", {
-            text "As long as Smoochum is Asleep, prevent all damage done to Smoochum by attacks."
-            delayedA {
-            }
-          }
-          move "Energy Antics", {
-            text "Move an Energy card attached to 1 of your opponent�s Pok�mon to another of your opponent�s Pok�mon. Smoochum is now Asleep."
-            energyCost ()
-            attackRequirement {}
-            onAttack {
-              damage 0
-            }
-          }
-
-        };
+        return copy (HeartgoldSoulsilver.SMOOCHUM_30, this);
       case LAPRAS_HGSS14:
         return basic (this, hp:HP090, type:WATER, retreatCost:2) {
           weakness M
           move "Ferry", {
             text "Search your discard pile for a Supporter card, show it to your opponent, and put it into your hand."
             energyCost W
-            attackRequirement {}
+            attackRequirement {
+              assert my.discard.any { it.cardTypes.is(SUPPORTER) } : "No Supporters in discard pile"
+            }
             onAttack {
-              damage 0
+              my.discard.findAll {
+                it.cardTypes.is(SUPPORTER)
+              }.select(text).moveTo(my.hand)
             }
           }
           move "Surf", {
-            text "40 damage. "
+            text "40 damage."
             energyCost W, W, C
             attackRequirement {}
             onAttack {
-              damage 0
+              damage 40
             }
           }
-
         };
       case SHUCKLE_HGSS15:
         return basic (this, hp:HP060, type:FIGHTING, retreatCost:1) {
           weakness W
           pokeBody "Fermenting Liquid", {
             text "Whenever you attach an Energy card from your hand to Shuckle, draw a card."
             delayedA {
+              after ATTACH_ENERGY, self, {
+                if (ef.reason == PLAY_FROM_HAND && ef.card.cardTypes.is(ENERGY)) {
```

Comment:
Isn't the card type already implied by what effect this is? Special/Basic specificity would be where you'd want to check, but all cards attached by ATTACH_ENERGY are ENERGY